rss/serializers/rss_2_0.py

```diff
@@ -0,0 +1,86 @@
+"""Serializer for RSS 2.0"""
+
+from typing import Tuple, Optional
+from arxiv import status
+from rss.serializers.serializer import Serializer
+from elasticsearch_dsl.response import Response
+import datetime
+from rfeed import *
+
+
+# rfeed Extensions are used to add namespaces to the rss element.
+class Content(rfeed.Extension):
+    def get_namespace(self):
+        return {"xmlns:content": "http://purl.org/rss/1.0/modules/content/"}
+
+
+class Taxonomy(rfeed.Extension):
+    def get_namespace(self):
+        return {"xmlns:taxo": "http://purl.org/rss/1.0/modules/taxonomy/"}
+
+
+class Syndication(rfeed.Extension):
+    def get_namespace(self):
+        return {"xmlns:syn": "http://purl.org/rss/1.0/modules/syndication/"}
+
+
+class Admin(rfeed.Extension):
+    def get_namespace(self):
+        return {"xmlns:admin": "http://webns.net/mvcb/"}
+
+
+class Rss_2_0(Serializer):
+
+    # TODO - Where to get correct value for pubDate?
+    # TODO - Is it OK to let rfeed add elements for "generator" and "docs"?
+    def get_xml(self, response: Response) -> Tuple[Optional[dict], int]:
+        """
+        Serializes the provided response data into RSS, version 2.0.
+
+        Parameters
+        ----------
+        response : Response
+            The search response data to be serialized.
+
+        Returns
+        -------
+        data : str
+            The serialized XML results.
+        status
+            The HTTP status code for the operation.
+        """
+
+        # Get the archive info from the first hit.  Is this OK?
+        archive = response.hits[0]["primary_classification"]["archive"]
+        feed = Feed(
+            title=archive["id"] + " updates on arXiv.org",
+            link="http://arxiv.org/",
```

Comment:
We'll want to make this configurable. There are two ways that I can think to do this off the top of my head:

1. Use a config parameter in ``config.py``. For example:

```python
# in config.py:
import os
FEED_BASE_URL = os.environ.get('FEED_BASE_URL', 'https://arxiv.org/')

# in your serializer module:
from arxiv.base.globals import get_application_config
...
    link=get_application_config()['FEED_BASE_URL]
```
The advantage is that it's pretty easy to change globally.

2. Extend your serializer to accept a base URL parameter, e.g. 

```python
class YourSerializer:
    def __init__(self, root_url):
        self._root_url = root_url
    
    ...
        link=self._root_url

# in routes.py:
    controller.get_xml(..., root_url=request.root_url)

# in controller:
    serializer = YourSerializer(root_url)
```
The advantage here is that it is set dynamically based on the request context (e.g. if the same app deployment is being served at two different URLs). It's also a little clearer what's going.

Reply:
Created ticket 1678 for this.